data/projects/vulture.md

```diff
@@ -1,57 +1,105 @@
-
-This project's aim is to provide such an auto removal feature via a coala bear,
-using the information vulture offers.
-
-To reach this goal the VultureBear should be modified to support removing a
-source range that vulture declared dead code.
-
-To realize this, some work has to be done upstream at vulture. Namely:
-
-- Make vulture available as a library for easier use for the bear
-- Implement a confidence value for results. This is necessary as vulture
-  produces false positives so a threshold has to be set what to accept for auto
-  removal.
-- Make vulture offer source ranges for the dead code. Right now vulture only
-  offers the beginning of the dead code, which makes removal hard. This could
-  possibly be done by removing everything up to the next node in the python AST.
-  Another possibility is that the enhanced AST of pyflakes could offer the
-  necessary information.
-- In case no way to offer source ranges is found, additional detection cases
-  can be implemented for vulture. For example identifying `if False` cases.
+#### Overview
+The motivation, here, is to extend the functionality of vulture as a library, and to pass on all metadata through the API
+and then to harness this utility in VultureBear for auto removing dead code, which would greatly optimise the bear. The
+second part of this project focuses on offering the source range of the dead code which would make auto-removal much
+easier. As of now, vulture only supplies the beginning of the dead code. Also, it proposes to enhance vulture in order to
+detect unreachable code (like `if False`, `if True else`, any code written after return statements, etc). - this shall help
+the user in trimming down their codebase without affecting usability. Also, the third part would be to implement a
+confidence value for every result, this shall be helpful when tackling false positives.
+
+![coala-VultureBear-Integration-picture](https://cloud.githubusercontent.com/assets/15556382/26275557/580cae2c-3d81-11e7-89ed-ac1ccbf9dc26.png)
+
+#### Goals
+- Modify vulture, such as to extend its core functionality as a library.
+- Refactor VultureBear accordingly for ensuring optimal performance.
+- Implement a method to acquire the source range of dead code and make suitable changes in the API and Bear.
+- Detect the instances of unreachable code, like `if False` statements
+- Analyse and implement a confidence value for results.
+
+#### Specifications
+##### 1.) Realise vulture’s API in VultureBear
+
+- Extending vulture’s API: This would allow the user to find all the unused code through a single abstract layer: ```get_unused_code```. Strategy here would be to:
+
+	* Parse all files straight away from dict(filename: filecontent)  - This would highly improve the performance of the bear later due to the time we save of memory copying.
+	* Return a sorted list of tuples [(item.filename, item.lineno, item.typ, item)...] - which would be easily configurable.
```

Comment:
I think you should try to not change vulture at all for this part. Try to change VultureBear to use vulture as a library without changing vulture. If we notice that vulture is missing some function, we can implement it later.